WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerSearchView.java

```diff
@@ -64,9 +63,7 @@ public boolean onMenuItemActionCollapse(MenuItem item) {
     public void enableSearchMode() {
         mMenuEdit.setVisible(false);
         mSitePickerActivity.setIsInSearchModeAndNullifyAdapter(true);
-        SitePickerSearchAdapter adapter = (SitePickerSearchAdapter) mSitePickerActivity.getAdapter();
-        mSitePickerActivity.getRecycleView().swapAdapter(adapter, true);
-        adapter.loadSites();
+        mSitePickerActivity.getRecycleView().swapAdapter(mSitePickerActivity.getAdapter(), true);
```

Comment:
This tight coupling between `SitePickerSearchView` and `SitePickerActivity` is a bit messy. Is there a way `SitePickerSearchView` can be written so it doesn't hold a reference to `SitePickerActivity`?